src/sync-controller.js

```diff
@@ -355,12 +355,12 @@ export default class SyncController extends videojs.EventTarget {
       this.logger_('tsO:', segmentInfo.timestampOffset);
 
       mappingObj = {
-        time: segmentInfo.timestampOffset,
-        mapping: segmentInfo.timestampOffset - timingInfo.start
+        time: segmentInfo.startOfSegment,
+        mapping: segmentInfo.startOfSegment - timingInfo.start
```

Comment:
I think to ensure we don't introduce a regression in the future, we need a unit test around modification of `timestampOffset` (which we do in mp4Probe) to make sure it doesn't change the start and end time of the segment.